jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1845,10 +1847,52 @@ public static URI toURI(String resource)
     {
         Objects.requireNonNull(resource);
 
-        // Only try URI for string for known schemes, otherwise assume it is a Path
-        return (ResourceFactory.isSupported(resource))
-            ? correctFileURI(URI.create(resource))
-            : Paths.get(resource).toUri();
+        if (URIUtil.hasScheme(resource))
+        {
+            try
+            {
+                URI uri = new URI(resource);
+
+                if (ResourceFactory.isSupported(uri))
+                    return correctFileURI(uri);
+
+                // We don't have a supported URI scheme
+                if (uri.getScheme().length() == 1)
+                {
+                    // Input is a possible Windows path disguised as a URI "D:/path/to/resource.txt".
+                    try
+                    {
+                        return toURI(Paths.get(resource).toUri().toASCIIString());
+                    }
+                    catch (InvalidPathException ignore)
+                    {
+                        // ignore
+                    }
+                }
+
+                // If we reached this point, that means the input String has a scheme,
+                // and is not recognized as supported by the registered schemes in ResourceFactory.
+                throw new IllegalArgumentException("URI scheme not supported: " + resource);
+            }
+            catch (URISyntaxException ignore)
+            {
+                // We have an input string that has what looks like a scheme, but isn't a URI.
+                // Eg: "C:\path\to\resource.txt"
+            }
+        }
+
+        // If we reached this point, we have a String with no valid scheme.
+        // Treat it as a Path, as that's all we have left to investigate.
+        try
+        {
+            return toURI(Paths.get(resource).toUri().toASCIIString());
+        }
+        catch (InvalidPathException ignore)
+        {
+            // ignore
+        }
+
+        throw new IllegalArgumentException("Unrecognized resource string: " + resource);
```

Comment:
We should include one or more of the exceptions above as the cause.
Also, perhaps we do not include the resource itself in the message as that might be user supplied data.

Reply:
Turns out those "exceptions above" often contain the resource string in their raw form as an Exception message.
I'm abandoning the idea of using those as suppressed exception for the final one.